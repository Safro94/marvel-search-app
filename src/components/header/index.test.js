/* eslint-disable import/first */
import React from 'react';

import { render, wait, waitForElement, fireEvent } from '@testing-library/react';

import Header from './';

jest.mock('../../components/shared/image', () => 'image');
jest.mock('./search', () => 'search');
jest.mock('../favourites', () => 'favourites');
jest.mock('../theme', () => 'theme');

jest.mock('./style', () => {
  return {
    Header: 'header',
    ImageContainer: 'imageContainer',
    Separator: 'separator',
    FavouriteContainer: 'FavouriteContainer',
  };
});

jest.mock('../../hooks/application', () => ({
  useApplication: jest.fn(() => ({
    setApplication: jest.fn(),
  })),
}));

describe('<Header /> ', () => {
  beforeEach(() => (history.length = 0));
  const history = [];

  it('It should render OK', async () => {
    // Act
    const { container } = render(<Header.WrappedComponent history={history} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('When the image is clicked, it should go to the home page', async () => {
    const url = '/';

    // Act
    const { getByTestId } = render(<Header.WrappedComponent history={history} />);
    const image = await waitForElement(() => getByTestId('image'));
    fireEvent.click(image);

    // Assert
    expect(history[0]).toBe(url);
  });
});
