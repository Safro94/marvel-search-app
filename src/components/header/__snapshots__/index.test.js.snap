// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header />  It should render OK 1`] = `
<div>
  <header>
    <imagecontainer>
      <image
        cursor="pointer"
        data-testid="image"
        height="30px"
        src="/marvel.jpg"
        width="60px"
      />
      <separator />
    </imagecontainer>
    <search />
    <favouritecontainer>
      <favourites />
      <theme />
    </favouritecontainer>
  </header>
</div>
`;
